src/main/java/AutoLotto/model/LottoMachine.java

```diff
@@ -3,67 +3,80 @@
 import autoLotto.exception.PurchaseException;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 public class LottoMachine {
     private static final String INVALID_PURCHASE_AMOUNT = "구입하신 수동 로또의 개수가 전체 로또 개수보다 많습니다.\n관리자에게 문의하여 로또를 재구매하시기 바랍니다.";
+    private final LottoGenerator lottoGenerator = new LottoGenerator();
 
     private List<Lotto> lottos;
 
-    public LottoMachine(List<Lotto> manualLottos, int numberOfTotalLottos) {
-        this.lottos = generateLottos(manualLottos, numberOfTotalLottos);
+    public LottoMachine(List<String> manualLottos, LottoPurchaseCount purchaseCount) {
+        this.lottos = generateLottos(manualLottos, purchaseCount);
     }
 
-    private List<Lotto> generateLottos(List<Lotto> manualLottos, int numberOfTotalLottos) {
-        validatePurchaseAmount(manualLottos, numberOfTotalLottos);
-        List<Lotto> lottos = setLottos(manualLottos, numberOfTotalLottos);
-        return lottos;
+    private List<Lotto> generateLottos(List<String> manualLottos, LottoPurchaseCount purchaseCount) {
+        validatePurchaseAmount(manualLottos, purchaseCount);
+        return setLottos(manualLottos, purchaseCount);
     }
 
-    private List<Lotto> setLottos(List<Lotto> manualLotto, int numberOfTotalLottos) {
+    private List<Lotto> setLottos(List<String> manualLotto, LottoPurchaseCount purchaseCount) {
         List<Lotto> lottos = new ArrayList<>();
-        int numberOfAutoLottos = numberOfTotalLottos - manualLotto.size();
 
-        lottos.addAll(setManualLottos(manualLotto));
-        lottos.addAll(buyAutoLottos(numberOfAutoLottos, new RandomLottoGeneratorStrategy()));
+        lottos.addAll(buyManualLottos(manualLotto, new ManualLottoGeneratorStrategy()));
+        lottos.addAll(buyAutoLottos(purchaseCount, new RandomLottoGeneratorStrategy()));
 
         return lottos;
     }
 
-    private List<Lotto> setManualLottos(List<Lotto> manualLottos) {
+    private List<Lotto> buyManualLottos(List<String> manualLottos, LottoGeneratorStrategy lottoGeneratorStrategy) {
         if (manualLottos.size() == 0) {
             return new ArrayList<>();
         }
 
-        return manualLottos;
+        List<Lotto> lottos = new ArrayList<>();
+
+        for (String lotto : manualLottos) {
+            List<Integer> lottoAsListInteger = convertStringToIntegerList(lotto);
+            lottos.add(new Lotto(lottoGenerator.generateManualLottoNumbers(lottoGeneratorStrategy, lottoAsListInteger)));
```

Comment:
lottoGenerator 이름부터 로또 생성기이니 generator 내부에서 Lotto 객체를 생성해서 반환해도 좋을 것 같아요
```suggestion
            lottos.add(lottoGenerator.generateManualLottoNumbers(lottoGeneratorStrategy, lottoAsListInteger));
```